src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1125,6 +1151,29 @@ public Collection<Object> createComponents(
             namedXContentRegistry.get()
         );
 
+        final var resourceSharingIndex = ResourceSharingConstants.OPENSEARCH_RESOURCE_SHARING_INDEX;
+        ResourceSharingIndexHandler rsIndexHandler = new ResourceSharingIndexHandler(resourceSharingIndex, localClient, threadPool);
+
+        ResourceAccessHandler resourceAccessHandler = new ResourceAccessHandler(threadPool, rsIndexHandler, adminDns, resourcePluginInfo);
+        // Resource Sharing index is enabled by default
+        boolean isResourceSharingEnabled = settings.getAsBoolean(
+            FeatureConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED,
+            FeatureConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED_DEFAULT
+        );
+        resourceSharingIndexManagementRepository = ResourceSharingIndexManagementRepository.create(
```

Comment:
Is this wrapper class needed? Looks like you can call `this.resourceSharingIndexHandler.createResourceSharingIndexIfAbsent` directly down below. What is the purpose of the callable as an arg to that method?

Reply:
I've followed suit with existing pattern: https://github.com/opensearch-project/security/blob/7603dc28a233a7df72f86537fb11d674f92f5155/src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java#L320

This will allows us to separate any index management tasks to its own class. Callable was a placeholder. I have removed extra abstraction since the only method offer by the repository was to create the index. I've replaced it with direct call